packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -165,15 +167,15 @@ export default class InternalModel {
   declare _retainedManyArrayCache: ConfidentDict<ManyArray>;
   declare _relationshipPromisesCache: ConfidentDict<RSVP.Promise<any>>;
   declare _relationshipProxyCache: ConfidentDict<PromiseManyArray | PromiseBelongsTo>;
-  declare currentState: any;
   declare error: any;
 
   constructor(public store: CoreStore | Store, public identifier: StableRecordIdentifier) {
     if (HAS_MODEL_PACKAGE) {
       _getModelPackage();
     }
-    this._tag = 0;
     this._id = identifier.id;
+    this._tag = 0;
```

Comment:
Can you please add a comment for why the weirdness with the id and why we need the tag